java-checks/src/main/java/org/sonar/java/checks/maven/helpers/MavenDependencyAbstractMatcher.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.maven.helpers;
+
+import org.sonar.maven.model.LocatedAttribute;
+import org.sonar.maven.model.maven2.Dependency;
+
+import javax.annotation.Nullable;
+
+import java.util.regex.Pattern;
+
+public abstract class MavenDependencyAbstractMatcher {
```

Comment:
Hierarchy of classes seems odd in this case as version and name of depednency have nothin in common and should not match on same parameter